plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetastoreAccessOperations.java

```diff
@@ -0,0 +1,198 @@
+                .setCatalog("iceberg")
+                .setSchema("test_schema")
+                .build();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session)
+                // Tests that inspect MBean attributes need to run with just one node, otherwise
+                // the attributes may come from the bound class instance in non-coordinator node
+                .setNodeCount(1)
+                .build();
+
+        File baseDir = queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data").toFile();
+        metastore = createCountingAccessesFileHiveMetaStore(baseDir);
+        queryRunner.installPlugin(new TestingIcebergPlugin(metastore, true));
+        queryRunner.createCatalog("iceberg", "iceberg");
+
+        queryRunner.execute("CREATE SCHEMA test_schema");
+        return queryRunner;
+    }
+
+    @BeforeMethod
+    public void prepare()
+    {
+        metastore.resetCounters();
+    }
+
+    @Test
+    public void testCreateTable()
+    {
+        assertUpdate("CREATE TABLE test_create (id VARCHAR, age INT)");
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 2,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+    }
+
+    @Test
+    public void testCreateTableAsSelect()
+    {
+        assertUpdate("CREATE TABLE test_ctas AS SELECT 1 AS age", 1);
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 2,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+    }
+
+    @Test
+    public void testSelect()
+    {
+        assertUpdate("CREATE TABLE test_select_from (id VARCHAR, age INT)");
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 2,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+
+        getDistributedQueryRunner().execute(TEST_SESSION, "SELECT * FROM test_select_from");
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
+                        GET_TABLE, 14,
+                        GET_ALL_DATABASES, 0,
+                        GET_TABLE_STATISTICS, 0));
+    }
+
+    @Test
+    public void testSelectWithFilter()
+    {
+        assertUpdate("CREATE TABLE test_select_from_where AS SELECT 2 as age", 1);
+        assertEquals(metastore.getMethodInvocations(),
+                ImmutableMap.of(GET_DATABASE, 1,
+                        GET_TABLE_WITH_PARAMETER, 0,
```

Comment:
let's not write expected entries with 0 invocations.
let's have non-zero only

(switching to multiset, or resetting the map  to empty -- both will give you that)

Reply:
ok